### PR TITLE
[2.26.x] Add default sort-by for ResultIterable

### PR DESCRIPTION
**Pull request by glenhein**
#### What does this PR do?

ResultIterable can return inconsistent results when paging through Solr results and the sort-by is  not used or when the sort-by field contains identical values. This PR sets the default sort-by to be by the metacard ID.

#### What are the relevant tickets?
Fixes: #____

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by glenhein** [comment]
build now

**Review by alexabird — APPROVED** [review]

File: catalog/core/catalog-core-impl/pubsub/pom.xml [hunk 1]
```diff
@@ -119,6 +119,7 @@
                             ddf.catalog.operation.impl,
                             ddf.catalog.data.impl.*,
                             ddf.catalog.util.impl,
+                            ddf.catalog.filter.impl,
```
Comment (jlcsmith) [inline catalog/core/catalog-core-impl/pubsub/pom.xml:122]: curious - why was this needed?

**Review by jlcsmith — APPROVED** [review]

**Review by derekwilhelm — APPROVED** [review]

**Comment by glenhein** [comment]
build now

**Comment by jlcsmith** [comment]
build now
